modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionMetadata.scala

```diff
@@ -0,0 +1,21 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{TransactionFee, TransactionReference, TransactionSalt}
+
+import derevo.cats.show
+import derevo.circe.magnolia.customizableDecoder
+import derevo.derive
+
+case object ConstructionMetadata {
+
+  @derive(customizableDecoder, show)
+  case class TransactionMetadata(
+    source: Address,
+    lastReference: TransactionReference,
+    fee: TransactionFee,
+    salt: Option[TransactionSalt]
```

Comment:
`salt` is auto-generated and imo should stay like that. if I correctly understand the metadata, salt shouldn't be there. the only field that needs to be fetched "being online" is a transaction last reference. then, you can use it in offline environment to construct a transaction.